### DISCUSSION TITLE
docker image base os

### DISCUSSION THREAD

**Discussion by robot-ranger** [mtconnect/cppagent#476] [General]
What is the reasoning behind moving to ubuntu docker images? Is it worth the 200% size increase?

**Comment by wsobel** [comment]
There’s a bug in the latest docker buildx arm and intel emulators that fails when building. It worked using the Ubuntu. If someone can figure out why buildx fails, I’ll be happy to switch back. I have tried a lot of changes and cannot get buildx to work. Only build. BestW(Sent from mobile)On Jul 9, 2024, at 15:56, robot-ranger ***@***.***> wrote:﻿
What is the reasoning behind moving to ubuntu docker images? Is it worth the 200% size increase?

**Reply by robot-ranger** [reply]
I'll take a look 🫡

**Reply by robot-ranger** [reply]
Is there a particular documented bug that you can reference? Or do you just mean that something in buildx generally changed that caused the build to fail

**Comment by wsobel** [comment]
It’s only with alpine. Go to the docker/alpine directory and try to do a multiplatform build. Compiler crashes randomly. Tried different versions of compiler. I think it is the underlying emulator that is causing the problem. When you do a build without buildx it works with alpine. Only when you try to cross compile. I tried on intel and arm platforms and could not get it to work. Also fails on the GitHub action. Tried to force older versions of the emulator with mixed success. Didn’t really work. Finally gave up and addressed other bugs. Any help is greatly appreciated. I’ll be working on it this week when I get back home. I think it is a bug since Ubuntu works and nothing has changed. The build failed at different places each time so it wasn’t just one package either. Seemed random. Also tried setting cpu count to 1, but no diff. Sticky problem. (Sent from mobile)On Jul 9, 2024, at 19:16, robot-ranger ***@***.***> wrote:﻿
I'll take a look 🫡

**Comment by wsobel** [comment]
That’s what I think. Cannot find something in buildx docs. See what you find. They may have fixed it.

**Comment by robot-ranger** [comment]
i concur, it doesnt build. lol. whats weird is we are building alpine for arm in github actions and it works fine?

**Comment by wsobel** [comment]
In GitHub I switched to Ubuntu.

**Reply by robot-ranger** [reply]
yup, i see that

**Comment by wsobel** [comment]
I have been stumped by this. It does not make any sense unless they broke the emulator integration to buildx.

Any help appreciated,
- W

**Reply by robot-ranger** [reply]
yes, very odd. Where was it hanging for you? for me, it was failing on `RUN apk update` instantly, almost looked like it failed parsing the dockerfile and not even actually "building"

**Comment by robot-ranger** [comment]
i previously had no success building locally, but it is now building as i type after `apt upgrade docker`, now running buildx version v0.15.1 instead of v0.11.0.

ill let you know how it goes, then ill try it on GH on my fork

**Reply by robot-ranger** [reply]
im well into the actual conan build now...
